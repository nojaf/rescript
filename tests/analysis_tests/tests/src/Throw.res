exception MyCustomThingToThrow(string)
exception NoArgsToThrow

// let x = () => throw
//                    ^com