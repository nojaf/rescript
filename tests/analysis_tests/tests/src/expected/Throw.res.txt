Complete src/Throw.res 3:21
posCursor:[3:21] posNoWhite:[3:20] Found expr:[3:11->3:21]
posCursor:[3:21] posNoWhite:[3:20] Found expr:[3:17->3:21]
Pexp_ident thro:[3:17->3:21]
Completable: Cpath Value[thro]
Package opens Stdlib.place holder Pervasives.JsxModules.place holder
Resolved opens 1 Stdlib
ContextPath Value[thro]
Path thro
[{
    "label": "throw(MyCustomThingToThrow)",
    "kind": 12,
    "tags": [],
    "detail": "User-defined Exception",
    "documentation": null,
    "filterText": "throw",
    "insertText": "throw(MyCustomThingToThrow($0))",
    "insertTextFormat": 2
  }, {
    "label": "throw(NoArgsToThrow)",
    "kind": 12,
    "tags": [],
    "detail": "User-defined Exception",
    "documentation": null,
    "filterText": "throw",
    "insertText": "throw(NoArgsToThrow)",
    "insertTextFormat": 2
  }, {
    "label": "throw(Not_found)",
    "kind": 12,
    "tags": [],
    "detail": "Built-in Exception",
    "documentation": null,
    "filterText": "throw",
    "insertText": "throw(Not_found($0))",
    "insertTextFormat": 2
  }, {
    "label": "throw(Invalid_argument)",
    "kind": 12,
    "tags": [],
    "detail": "Built-in Exception",
    "documentation": null,
    "filterText": "throw",
    "insertText": "throw(Invalid_argument($0))",
    "insertTextFormat": 2
  }, {
    "label": "throw(Assert_failure)",
    "kind": 12,
    "tags": [],
    "detail": "Built-in Exception",
    "documentation": null,
    "filterText": "throw",
    "insertText": "throw(Assert_failure($0))",
    "insertTextFormat": 2
  }, {
    "label": "throw(Failure)",
    "kind": 12,
    "tags": [],
    "detail": "Built-in Exception",
    "documentation": null,
    "filterText": "throw",
    "insertText": "throw(Failure($0))",
    "insertTextFormat": 2
  }, {
    "label": "throw(Match_failure)",
    "kind": 12,
    "tags": [],
    "detail": "Built-in Exception",
    "documentation": null,
    "filterText": "throw",
    "insertText": "throw(Match_failure($0))",
    "insertTextFormat": 2
  }, {
    "label": "throw(Division_by_zero)",
    "kind": 12,
    "tags": [],
    "detail": "Built-in Exception",
    "documentation": null,
    "filterText": "throw",
    "insertText": "throw(Division_by_zero)",
    "insertTextFormat": 2
  }]

